Expecto/Expecto.fs

```diff
@@ -484,9 +485,37 @@ module Tests =
         ArgsUsage (Args.usage commandName options, errors)
 
   /// Prints out names of all tests for given test suite.
-  let listTests (join: JoinWith) test =
-    Test.toTestCodeList test
-    |> Seq.iter (fun t -> printfn "%s" (join.format t.name))
+  let listTests config test =
+    let toStateChar state =
+      match state with
+      | Normal  -> "N"
+      | Pending -> "P"
+      | Focused -> "F"
+
+    let tests =
+      Test.toTestCodeList test
+      |> Seq.filter(fun t -> List.isEmpty config.listStates ||
+                             List.contains t.state config.listStates)
+      |> Seq.map(fun t -> toStateChar t.state, config.joinWith.format t.name)
+      |> Seq.toList
+
+    let hideState = tests |> List.exists(fun (stateChar,_) -> stateChar <> "N") |> not
+
+    let result = System.Text.StringBuilder()
+    tests
+    |> Seq.iter (fun (stateChar, name) ->
+      if hideState then
+        Printf.bprintf result "\n%s" name
+      else
+        Printf.bprintf result "\n%s %s" stateChar name
+    )
+    Printf.bprintf result "\n"
+
+    logger.logWithAck Info (
+      Message.eventX "{result}"
```

Comment:
Better to make the event the result IMO;

Reply:
please explain, `Message.eventX result` without fields?